test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -39,35 +47,23 @@ namespace integration_framework {
   const std::string IntegrationTestFramework::kAdminId = "admin@test";
   const std::string IntegrationTestFramework::kAssetName = "coin";
 
+  IntegrationTestFramework::IntegrationTestFramework(
+      size_t maximum_proposal_size,
+      std::function<void(integration_framework::IntegrationTestFramework &)>
+          deleter)
+      : maximum_proposal_size_(maximum_proposal_size), deleter_(deleter) {}
+
   IntegrationTestFramework::~IntegrationTestFramework() {
-    pqxx::lazyconnection connection(iroha_instance_->pg_conn_);
-    const auto drop = R"(
-DROP TABLE IF EXISTS account_has_signatory;
-DROP TABLE IF EXISTS account_has_asset;
-DROP TABLE IF EXISTS role_has_permissions;
-DROP TABLE IF EXISTS account_has_roles;
-DROP TABLE IF EXISTS account_has_grantable_permissions;
-DROP TABLE IF EXISTS account;
-DROP TABLE IF EXISTS asset;
-DROP TABLE IF EXISTS domain;
-DROP TABLE IF EXISTS signatory;
-DROP TABLE IF EXISTS peer;
-DROP TABLE IF EXISTS role;
-DROP TABLE IF EXISTS height_by_hash;
-DROP TABLE IF EXISTS height_by_account_set;
-DROP TABLE IF EXISTS index_by_creator_height;
-DROP TABLE IF EXISTS index_by_id_height_asset;
-)";
-
-    pqxx::work txn(connection);
-    txn.exec(drop);
-    txn.commit();
-    connection.disconnect();
-
-    iroha::remove_dir_contents(iroha_instance_->block_store_dir_);
+    if (deleter_) {
+      deleter_(*this);
+    }
+    // the code below should be executed anyway in order to prevent app hang
+    if (iroha_instance_ and iroha_instance_->instance_) {
+      iroha_instance_->instance_->terminate();
+    }
   }
 
-  IntegrationTestFramework &IntegrationTestFramework::setInitialState(
+  shared_model::proto::Block IntegrationTestFramework::defaultBlock(
```

Comment:
Better to return const ref for interface here. There are no reasons to pass concrete implementation here.

Reply:
It is a static function, thus for your case there's a need in static object, that may be not a good idea

Reply:
Decided with @muratovv and @l4l left this place "as is".